pkg/api/api.go

```diff
@@ -59,9 +59,10 @@ type API struct {
 	// TweakResponseFunc can be optionally registered to add custom code that changes the
 	// payload of the downloaded api content (e.g. to exclude unwanted/unnecessary fields)
 	TweakResponseFunc func(map[string]any)
-	// Parent is used for SubPath APIs to store information about the configuration type and ID of the related
-	// configuration once Resolved() is called.
+	// Parent is used for SubPath APIs to store the related configuration.
 	Parent *API
+	// ParentObjectID is the parent object ID for a SubPath API once it has been applied.
+	ParentObjectID string
```

Comment:
@warber, what name ideas do you have?

Reply:
you could change `HasParent() bool` to `HasParent() (string,bool)` and make `api.ParentObjectID`  unexported.
If the change of the signature is annoying, you can introduce a `HasParentID() (string,bool)`function additionally and leave the other as is.
just an idea :)
otherwise, i think `ResolvedParentID`is good.

Reply:
I changed it to `AppliedParentObjectID` to make it clear which action it is the result of. Probably should un-export all these fields, but that would be beyond the scope of this PR. But we could / should follow up here.